### ISSUE TITLE
[LC0017] Analyzer crashes with InvalidCastException continued...

### ISSUE THREAD

**Issue by jwikman** [StefanMaron/BusinessCentral.LinterCop#156]
Similar to #149.

But I get another casting error, running BC.LinterCop v0.24.3.0

```
[{
	"resource": "/<path to my app>/App/app.json",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "AD0001",
	"severity": 4,
	"message": "Analyzer 'BusinessCentral.LinterCop.Design.Rule0017WriteToFlowField' threw an exception of type 
'System.InvalidCastException' with message 'System.InvalidCastException: Unable to cast object of type 
'Microsoft.Dynamics.Nav.CodeAnalysis.BoundLiteral' to type 'Microsoft.Dynamics.Nav.CodeAnalysis.IFieldAccess'.\r\n   at 
BusinessCentral.LinterCop.Design.Rule0017WriteToFlowField.CheckForWriteToFlowField(OperationAnalysisContext context)\r\n   at 
Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass53_1.<ExecuteOperationAction>b__1() in 
D:\\a\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 764\r\n   at 
Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, 
Action analyze, Nullable`1 info) in 
D:\\a\\1\\s\\source\\Prod\\Microsoft.Dynamics.Nav.CodeAnalysis\\DiagnosticAnalyzer\\AnalyzerExecutor.cs:line 1100'",
	"source": "AL",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 1
}]
```

**Comment by StefanMaron** [comment]
You should see a LC0000 info message which shows which line crashes the linter. Then I will be able to analyze this further

**Comment by jwikman** [comment]
Aaah, nice! 👍

I now have LC0000 in a dozen places, all seems to be on line 41:
```
 info LC0000: There was an Error in Rule "Rule0017" of type "InvalidCastException" at Line 41
```

**Comment by jwikman** [comment]
Some of the error info is now lost, though.
Like `Unable to cast object of type 
'Microsoft.Dynamics.Nav.CodeAnalysis.BoundLiteral' to type 'Microsoft.Dynamics.Nav.CodeAnalysis.IFieldAccess'` 
But maybe you do not need this when you've got the line number?

**Comment by StefanMaron** [comment]
The idea was that the info message will show which line of AL code leads to the error. Then I can reproduce and debug.

/Stefan
________________________________
Von: Johannes Wikman ***@***.***>
Gesendet: Friday, February 25, 2022 8:42:36 PM
An: StefanMaron/BusinessCentral.LinterCop ***@***.***>
Cc: Stefan Maroń ***@***.***>; Comment ***@***.***>
Betreff: Re: [StefanMaron/BusinessCentral.LinterCop] [LC0017] Analyzer crashes with InvalidCastException continued... (Issue #156)


Some of the error info is now lost, though.
Like Unable to cast object of type 'Microsoft.Dynamics.Nav.CodeAnalysis.BoundLiteral' to type 'Microsoft.Dynamics.Nav.CodeAnalysis.IFieldAccess'
But maybe you do not need this when you've got the line number?

**Comment by jwikman** [comment]
As long as you can repro from that, we're happy. 🙂

But if you need more hints on what really fails on a line, I guess that you could just output the exception message in the warning.

**Comment by StefanMaron** [comment]
The original error message is all I can get and it does not help me without the AL code causing it.
You said that you dont know what line is casing the error, so I added LC 0000.

If you can, please give me examples of AL code that leads to LC0000. Without that AL code samples I can not fix the Linter code

/Stefan
________________________________
Von: Johannes Wikman ***@***.***>
Gesendet: Friday, February 25, 2022 10:16:00 PM
An: StefanMaron/BusinessCentral.LinterCop ***@***.***>
Cc: Stefan Maroń ***@***.***>; Comment ***@***.***>
Betreff: Re: [StefanMaron/BusinessCentral.LinterCop] [LC0017] Analyzer crashes with InvalidCastException continued... (Issue #156)


As long as you can repro from that, we're happy. 🙂

But if you need more hints on what really fails on a line, I guess that you could just output the exception message in the warning.

**Comment by StefanMaron** [comment]
That should be fixed in the latest version

**Comment by jwikman** [comment]
I can confirm that this is solved in 0.24.5. 👍
